cassandra/metadata.py

```diff
@@ -348,7 +361,8 @@ def get_host(self, endpoint_or_address, port=None):
         if not isinstance(endpoint_or_address, EndPoint):
             return self._get_host_by_address(endpoint_or_address, port)
 
-        return self._hosts.get(endpoint_or_address)
+        host_id = self._host_id_by_endpoint.get(endpoint_or_address)
+        return self._hosts.get(host_id)
```

Comment:
reading without holding a lock

Reply:
also the original code, there wasn't a lock on it

```
return self._hosts.get(endpoint_or_address)
```

Reply:
it's a bug, you shouldn't access protected resources without holding a lock

Reply:
I'll add it, and let the tests run again